File: task3/C++/task3.cpp
//g++ task3.cpp -o task3.exe
//./task3.exe
#include <iostream>
#include <fstream>
#include <cmath>

using namespace std;

// проверка на полный квадрат
bool isPerfectSquare(int x) {
    int root = (int)sqrt(x);
    return root * root == x;
}

int main() {
    int choice;
    cout << "1 - input from console\n2 - input from file\n";
    cin >> choice;

    int count = 0;

    if (choice == 1) {
        int n;
        cout << "Enter N: ";
        cin >> n;

        int x;
        for (int i = 0; i < n; i++) {
            cin >> x;
            if (isPerfectSquare(x)) {
                count++;
            }
        }
    }
    else if (choice == 2) {
        string filename;
        cout << "Enter file name: ";
        cin >> filename;

        ifstream file(filename);
        if (!file.is_open()) {
            cout << "File open error\n";
            return 1;
        }

        int n, x;
        file >> n;

        for (int i = 0; i < n; i++) {
            file >> x;
            if (isPerfectSquare(x)) {
                count++;
            }
        }

        file.close();
    }
    else {
        cout << "Invalid choice\n";
        return 1;
    }

    cout << "Result: " << count << endl;

    return 0;
}

File: task3/Elixir/task3.exs
defmodule Main do
  # проверка на полный квадрат
  def is_square(x) do
    root = :math.sqrt(x) |> trunc()
    root * root == x
  end

  def main do
    IO.puts("1 - input from console\n2 - input from file")
    {choice, _} = Integer.parse(IO.gets("") |> String.trim())

    count =
      cond do
        choice == 1 ->
          IO.puts("Enter N:")
          {n, _} = Integer.parse(IO.gets("") |> String.trim())

          Enum.reduce(1..n, 0, fn _, acc ->
            {x, _} = Integer.parse(IO.gets("") |> String.trim())
            if is_square(x), do: acc + 1, else: acc
          end)

        choice == 2 ->
          IO.puts("Enter file name:")
          filename = IO.gets("") |> String.trim()

          case File.read(filename) do
            {:ok, content} ->
              nums =
                content
                |> String.split()
                |> Enum.map(&String.to_integer/1)

              [n | rest] = nums

              rest
              |> Enum.take(n)
              |> Enum.count(&is_square/1)

            _ ->
              IO.puts("File open error")
              0
          end

        true ->
          IO.puts("Invalid choice")
          0
      end

    IO.puts("Result: #{count}")
  end
end

Main.main()
